src/nvim/drawscreen.c

```diff
@@ -619,6 +621,8 @@ int update_screen(void)
   screen_search_hl.rm.regprog = NULL;
 
   FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
+    wp->w_buffer->b_signcols.resized = false;
```

Comment:
Actually it needs to stay `true` until the above loop has ended. Otherwise, subsequent encountered windows displaying the same buffer will not have their statuscolumn rebuild. I.e. this would fail:
https://github.com/neovim/neovim/blob/ca7f8786a0eb578895400e23cd21e25cc0f91800/test/functional/ui/statuscolumn_spec.lua#L838-L845

Could add a nested loop over all windows in `buf_signcols_validate()` and move it there to mark all windows holding the validated buffer.